client/js/table-view.js

```diff
@@ -2,146 +2,165 @@ const { getLetterRange } = require('./array-util');
 const { removeChildren, createTH, createTR, createTD } = require('./dom-util');
 
 class TableView {
-    constructor(model) {
-        this.model = model;
-    }
+  constructor(model) {
+    this.model = model;
+  }
+
+  init() {
+    this.initDomReferences();
+    this.initCurrentCell();
+    this.renderTable();
+    this.attachEventHandlers();
+  }
+
+  initDomReferences() {
+    this.headerRowEl = document.querySelector('THEAD TR');
+    this.sheetBodyEl = document.querySelector('TBODY');
+    this.footerRowEl = document.querySelector('TFOOT TR');
+    this.formulaBarEl = document.querySelector('#formula-bar');
+    this.addColumnButton = document.querySelector('#addColumn');
+    this.addRowButton = document.querySelector('#addRow');
+  }
+
+  initCurrentCell() {
+    this.currentCellLocation = { col: 0, row: 0 };
+    this.renderFormulaBar();
+  }
+
+  normalizeValueForRendering(value) {
+    return value || '';
+  }
+
+  renderFormulaBar() {
+    const currentCellValue = this.model.getValue(this.currentCellLocation);
+    this.formulaBarEl.value = this.normalizeValueForRendering(currentCellValue);
+    this.formulaBarEl.focus();
+  }
+
+  renderTable() {
+    this.renderTableHeader();
+    this.renderTableBody();
+    this.renderTableFooter();
+  }
+
+  renderTableHeader() {
+    // clear header row
+    removeChildren(this.headerRowEl);
+    // get letters and build elements
+    getLetterRange('A', this.model.numCols)
+      .map(colLabel => createTH(colLabel))
+      .forEach(th => this.headerRowEl.appendChild(th));
+  }
+
+
+  isCurrentCell(col, row) {
+    return this.currentCellLocation.row === row &&
+      this.currentCellLocation.col === col;
```

Comment:
Better readability: line up "this.currentCellLocation.col" with "this.currentCellLocation.row" above it (so that "this" is just underneath "this")